Automattic/jetpack
# Social:  Load share counter on the server side.

We are dispatching an action [in the front-end code ](https://github.com/Automattic/jetpack/blob/trunk/projects/plugins/social/src/js/components/header/index.js#:~:text=dispatch.getSharesCount()%3B )to get the share counter. This PR moved the logic to fetch the share counter to the backend. To facilitate this, we have moved tweaked logic in the Publicize package. 




#### Changes proposed in this Pull Request:

* Moved the call to `wpcom_json_api_request_as_blog` from `projects/packages/publicize/src/class-rest-controller.php` to `projects/packages/publicize/src/class-publicize-base.php`
* Added a function `get_publicize_shares_count` to `projects/packages/publicize/src/class-publicize-base.php`
* Called the above function from `projects/plugins/social/src/class-jetpack-social.php` to populate the initial state with the share counter
* Got rid of the dispatch actions on the front-end, since the data is loaded on the server side now. 

#### Jetpack product discussion


p1655976998473259-slack-C02JJ910CNL

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Make sure the share counter loads correctly.
* Without this branch, an API Call to `wp-json/jetpack/v4/publicize/shares-count?_locale=user` can be observed on the network debugger on the browser. With this brach checked out, you will no longer see the aforementioned network call being made.
